### PR TITLE
Allow Diagnostic-Id in Applicaiton or System properties

### PR DESCRIPTION
**Pull request by alrod**

### CODE REVIEW

File: src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubListener.cs [hunk 1]
```diff
@@ -315,13 +315,15 @@ private bool TryGetLinkedActivity(EventData message, out Activity link)
...
+                if ((message.SystemProperties.TryGetValue("Diagnostic-Id", out var diagnosticIdObj) || message.Properties.TryGetValue("Diagnostic-Id", out diagnosticIdObj)) 
+                    && diagnosticIdObj is string diagnosticIdString)
                 {
-                    return false;
+                    link = new Activity("Microsoft.Azure.EventHubs.Process");
+                    link.SetParentId(diagnosticIdString);
+                    return true;
                 }
 
-                link = message.ExtractActivity();
```
Comment (brettsam) [inline src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubListener.cs]: Do we not need to call `ExtractActivity` anymore?

Comment (lmolkova) [reply src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubListener.cs]: ExtractActivity is only available on Application-Properties, and changing old EventHub SDK does not seem reasonable option  as it's in the maintenance mode and Azure SDK is shipping [new flavor](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventhub/Azure.Messaging.EventHubs).

 I took the only part of `ExtractActivity` we care about (which is `activitiy.SetParentId`). Besides that  `ExtractActivity` also populates Baggage on the Activity which we drop for EventHub scenario anyway.

**Comment by alrod** [comment]
@lmolkova, some tests are broken, can you please take a look?

**Comment by lmolkova** [comment]
@alrod sorry about that, here is the fix https://github.com/lmolkova/azure-functions-eventhubs-extension/commit/f0d7080571fd3f72974cdb94af547a5f101c6b8d

**Comment by lmolkova** [comment]
test failure again, seems that test is flaky - fixing it

**Comment by lmolkova** [comment]
another fix: https://github.com/lmolkova/azure-functions-eventhubs-extension/commit/5dba6aac35cd7734a5b8ce284f36a74f5858e36a

**Comment by lmolkova** [comment]
@alrod can you update the branch with https://github.com/lmolkova/azure-functions-eventhubs-extension/commit/5dba6aac35cd7734a5b8ce284f36a74f5858e36a please?

**Pull request merged**